crud/common/utils.lua

```diff
@@ -375,6 +386,52 @@ function utils.get_bucket_id_fieldno(space, shard_index_name)
     return bucket_id_index.parts[1].fieldno
 end
 
+-- NOTE: it is like get_bucket_id_fieldno()
+-- but it uses space's format instead of index
+function utils.get_fieldno_by_name(space_format, field_name)
+    field_name = field_name or 'bucket_id'
+    local fieldno
+    for i, field_format in ipairs(space_format) do
+        if field_format.name == field_name then
+            fieldno = i
+            break
+        end
+    end
+    if fieldno == nil then
+        return nil, ShardingError:new('%q field is not found in a space format', field_name)
+    end
+
+    return fieldno
+end
+
+--- Get a map with fieldno of passed field's names.
+--
+-- @function get_keys_fieldno_map
+--
+-- @param table space_format
+--  A space format
+--
+-- @param keys
+--  A table with field names.
+--
+-- @return[1] table
+-- @treturn[2] nil
+-- @treturn[2] table Error description
+--
+function utils.get_keys_fieldno_map(space_format, field_names)
+    dev_checks('table', 'table')
+    local t = {}
+    for _, field_name in ipairs(field_names) do
+        local fieldno, err = utils.get_fieldno_by_name(space_format, field_name)
+        if fieldno == nil then
+            return {}, err
+        end
+        t[fieldno] = true
+    end
```

Comment:
```lua
--- a/crud/common/utils.lua
+++ b/crud/common/utils.lua
@@ -386,22 +386,13 @@ function utils.get_bucket_id_fieldno(space, shard_index_name)
     return bucket_id_index.parts[1].fieldno
 end
 
--- NOTE: it is like get_bucket_id_fieldno()
--- but it uses space's format instead of index
-function utils.get_fieldno_by_name(space_format, field_name)
-    field_name = field_name or 'bucket_id'
-    local fieldno
+function utils.get_format_fieldno_map(space_format)
+    local t = {}
     for i, field_format in ipairs(space_format) do
-        if field_format.name == field_name then
-            fieldno = i
-            break
-        end
-    end
-    if fieldno == nil then
-        return nil, ShardingError:new('%q field is not found in a space format', field_name)
+        t[field_format.name] = i
     end
 
-    return fieldno
+    return t
 end
 
 --- Get a map with fieldno of passed field's names.
@@ -416,15 +407,15 @@ end
 --
 -- @return[1] table
 -- @treturn[2] nil
--- @treturn[2] table Error description
 --
 function utils.get_keys_fieldno_map(space_format, field_names)
     dev_checks('table', 'table')
     local t = {}
+    local fieldno_map = utils.get_format_fieldno_map(space_format)
     for _, field_name in ipairs(field_names) do
-        local fieldno, err = utils.get_fieldno_by_name(space_format, field_name)
+        local fieldno = fieldno_map[field_name]
         if fieldno == nil then
-            return {}, err
+            return nil
         end
         t[fieldno] = true
     end
```